x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -222,34 +226,32 @@ public String toString() {
         }
 
         /**
-         * Generate positive test cases for binary functions that operate on an {@code numeric}
+         * Generate positive test cases for unary functions that operate on an {@code numeric}
          * fields by casting them to {@link DataTypes#DOUBLE}s.
          */
         public static List<TestCaseSupplier> forUnaryCastingToDouble(String name, String argName, DoubleUnaryOperator expected) {
```

Comment:
thought (highly out of scope for this PR): I wonder if e.g. refactors will be harder in the future because of the fact that `AbstractFunctionTestCase` has two separate responsibilities at the moment, namely
- serving as base class for function tests, and
- controlling how docs for functions are being auto-generated.

I imagine that we'd maybe want to separate this in the future. For instance, we could add a method to our function classes that returns the allowed type combinations + result; then `AbstractFunctionTestCase` could generate negative tests for any type combinations that were not marked as allowed, and a separate class could generate docs from the allowed type combinations+results.

Reply:
I suppose this could become a problem in a few ways:
1. We want to document type combinations we aren't testing.
2. We want to test combinations we don't document
3. Everything just gets gummed up together

I think 1 is *fairly* unlikely and generally a bad thing to want. We already have 2 and have some work arounds for it, but they aren't really "pluggable", just heuristics like "the number of arguments is the same". That seems ok for now.

3 is a bigger concern, though one I think I'd like to ignore until it actually happens.

Reply:
I agree that we want 1 and 2, and am a bit afraid of 3; I think we're already doing the right thing, I'm just wondering if we can avoid gumming up by having our functions actively declare what type combos are supported, rather than inferring the supported types implicitly from the tests; we can still ensure 1 and 2 this way, I think.